tdrs-backend/tdpservice/users/api/middleware.py

```diff
@@ -0,0 +1,34 @@
+"""Handle authorization calls globally."""
+import datetime
+from django.conf import settings
+
+
+class AuthUpdateMiddleware:
+    """Update authorization cookie with new requests."""
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """Update cookie."""
+        response = self.get_response(request)
+        now = datetime.datetime.now()
+        timeout = now + datetime.timedelta(minutes=settings.SESSION_TIMEOUT)
+
+        # if there is no user, the user is currently
+        # in the authentication process so we can't
+        # update the cookie yet
+        if request.user.is_authenticated:
+            id_token = request.COOKIES.get("id_token")
+            response.set_cookie(
+                "id_token",
+                value=id_token,
+                max_age=None,
+                expires=timeout,
+                path="/",
+                domain=None,
+                secure=True,
+                httponly=True,
```

Comment:
https://docs.djangoproject.com/en/3.1/ref/request-response/#django.http.HttpResponse.set_cookie

Setting `httponly=True` is a good security move. Should we set `samesite='Strict'`?

Reply:
@carltonsmith Or would setting `samesite='Strict'` break things given that the frontend/backend are different apps? This also makes me think about our CSP settings.